crates/proto/src/domain/mod.rs

```diff
@@ -9,18 +9,20 @@ pub mod transactions;
 // UTILITIES
 // ================================================================================================
 
-pub fn convert<T, From, To>(from: T) -> Vec<To>
+pub fn convert<T, From, To, R>(from: T) -> R
 where
     T: IntoIterator<Item = From>,
     From: Into<To>,
+    R: FromIterator<To>,
 {
     from.into_iter().map(|e| e.into()).collect()
 }
```

Comment:
Similar to one of the previous comments, maybe this could be something like:
```rust
pub fn convert<From, To>(from: impl IntoIterator<Item = From>) -> impl Iterator<Item = To>
where
    From: Into<To>
```
This way, the caller would be responsible for calling `.collect()` or could use the returned iterator in some other ways.

Though, I'm wondering if we are getting much convenience from this function at all (and maybe it make sense to remove it). Though, this would probably be for another PR.

Reply:
I suspect removal and handing control back to the caller ito allocations and `iter().map(..)` chaining makes the most sense. Though I don't have the full sense of the ergonomics here yet.